docs/plain-bundle-spec.md

```diff
@@ -0,0 +1,138 @@
+the [plain bundle provisioner](/internal/provisioner/plain/README.md) supports.
+
+The currently implemented plain bundle format is the `plain+v0` format. The name of the bundle format, `plain+v0`
+combines the type of bundle (plain) with the current schema version (v0).
+The [plain bundle provisioner](/internal/provisioner/plain/README.md) is able to source `plain+v0` bundles and install
+them onto a Kubernetes cluster.
+
+> Note: the plain+v0 bundle format is at schema version v0, which means it's an experimental format that is subject
+> to change.
+
+Supported source types for a plain bundle currently include the following:
+
+* A container image
+* A directory in a git repository
+
+Additional source types, such as a local volume or a generic URI-based resource, are on the roadmap. These source types
+all present the same content, a directory containing static Kubernetes YAML manifests, in a different ways.
+
+## Example
+
+For example, below is a minimal example of a Dockerfile that builds a `plain+v0` image-based bundle from a directory
+containing static Kubernetes manifests.
+
+```dockerfile
+FROM scratch
+COPY /manifests /manifests
+```
+
+where the given `manifests` directory contains the Kubernetes resources required to deploy an application, for example:
+
+```
+manifests
+├── namespace.yaml
+├── cluster_role.yaml
+├── role.yaml
+├── serviceaccount.yaml
+├── cluster_role_binding.yaml
+├── role_binding.yaml
+└── deployment.yaml
+```
+
+For an example of a directory backing a git-based bundle,
+see [the combo manifests directory](https://github.com/operator-framework/combo/tree/main/manifests). Any directory that
+contains only static Kubernetes manifests checked into a git repository accessible via a remote URL can be considered a
+plain bundle and sourced by the plain provisioner.
+
+> Note: there must be at least one resource in the manifests directory in order for the bundle to be a valid
+> plain+v0 bundle.
+
+## Technical Details
+
+* The static manifests must be located in the root-level /manifests directory in a bundle image for the bundle to be a
+  valid `plain+v0` bundle that the provisioner can unpack. A plain bundle image without a /manifests directory is
+  invalid and will not be successfully unpacked onto the cluster. For a git-based plain bundle, this limitation does not
+  exist, and the manifests can be in any directory in the repository. The manifest directory is assumed to be
+  ./manifests in a git-based plain bundle but can be overwritten.
+* The /manifests directory may contain additional sub-directories which contain static manifests.
+* The plain image-based bundle can be built from any base image, but `scratch` is recommended as it keeps the resulting
+  bundle image a minimal size.
+* Additional bundle metadata can be provided in two ways: either via image labels or by including a `/metadata`
+  directory inside the bundle. For a `plain+v0` bundle either is acceptable and entirely optional.
```

Comment:
IMO, we should not allow metadata to be provided in different ways. That introduces unnecessary ambiguity and complexity:
- do I have to include both?
- if I include both, which takes precedence?
- is precedence always the same no matter the context (e.g. building a catalog vs unpacking the bundle on cluster)?

My vote: Only metadata on the filesystem. That ensures that all filesystem-based sources can provide metadata in a consistent way. The downside is that for images, you have to pull the image to get metadata, rather than just requesting labels. But IMO the tradeoff is easily worth it, at least given my personal experience and confusion with registry+v1 metadata.

For bundle metadata, I think we should define a single file (`bundle.yaml`) that must exist in the root of _ALL_ bundle directories for all bundles types (that would mean we'd need special handling for registry+v1 bundles). And the only required fields would be `apiVersion` and `kind`.  That way any provisioner can look at any bundle and immediately parse this one file to definitely identify what it is looking at.

Reply:
Do we even needs to document that we have support (or rather, will have support down the line) for exposing bundle metadata? Can this be added over time once we have a more concrete use case for bundle-level metadata?

Reply:
Yeah, if we completely remove this piece, that also satisfies me for now.